x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/DataFrameDataExtractor.java

```diff
@@ -426,5 +441,9 @@ public boolean isTraining() {
         public int getChecksum() {
             return Arrays.hashCode(values);
         }
+
+        public String getSortKey() {
+            return (String) hit.getSortValues()[0];
```

Comment:
Is `hit.getSortValues()` always non-empty?

Reply:
Yes because we own the search from within the extractor and we are sorting on the id copy field.